crates/core_types/src/network/entry_aspect.rs

```diff
@@ -95,12 +97,24 @@ impl EntryAspect {
     }
     /// NB: this is the inverse function of entry_to_meta_aspect,
     /// so it is very important that they agree!
-    pub fn entry_address(&self) -> &Address {
-        match self {
-            EntryAspect::LinkAdd(link_data, _) => link_data.link.base(),
-            EntryAspect::LinkRemove((link_data, _), _) => link_data.link.base(),
-            _ => self.header().entry_address(),
-        }
+    pub fn entry_address(&self) -> Result<Address, HolochainError> {
+        Ok(match self {
+            EntryAspect::Content(_, header) => match header.link_update_delete() {
+                Some(ref updated_entry) => updated_entry.clone(),
+                None => header.entry_address().clone(),
+            },
+            EntryAspect::LinkAdd(link_data, _) => link_data.link.base().clone(),
+            EntryAspect::LinkRemove((link_data, _), _) => link_data.link.base().clone(),
+            EntryAspect::Update(_, header) | EntryAspect::Deletion(header) => {
+                header.link_update_delete().ok_or_else(|| {
+                    HolochainError::ErrorGeneric(format!(
+                        "no link_update_delete on Update/Deletion entry header. Header: {:?}",
+                        header
+                    ))
+                })?
+            }
+            EntryAspect::Header(header) => header.entry_address().clone(),
```

Comment:
Is this correct?  Is entry address of a header aspect supposed to be the address of the entry?  Also I note that there isn't an inverse for the Header aspect in `entry_to_meta_aspect` so this worries me that we have it here.

Reply:
You're right, it's not correct. There's no inverse I guess because Header is not a "meta" aspect. And EntryAspect::Header is unused, anyway. I fixed it here though.